File: kotlin-formal/Java/KotlinBaseLexer.java
import org.antlr.v4.runtime.Lexer;
import org.antlr.v4.runtime.*;

public class KotlinBaseLexer extends Lexer {

    public KotlinLexer(TokenStream input) {
            super(input);
            _interp = new LexerATNSimulator(this,_ATN,_decisionToDFA,_sharedContextCache);
        }

    public void PopModeIfNecessary(){
        if (!_modeStack.isEmpty()) {
            popMode();
        }
    }
}

File: kotlin-formal/CSharp/KotlinBaseLexer.cs
using Antlr4.Runtime;

public abstract class KotlinBaseLexer : Lexer
{
    public void PopModeIfNecessary()
    {
        if (_modeStack.Count > 0)
            PopMode();
    }

    protected KotlinBaseLexer(ICharStream input) : base(input)
    {
    }

}